spec.emu

```diff
@@ -931,10 +945,11 @@
             Line `;` LineList
         </emu-grammar>
         <emu-alg>
-          1. Perform DecodeMappingsField of |Line| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. Perform DecodeMappingsField of |Line| with arguments _state_, _mappings_, _names_<ins>,</ins> <del>and</del> _sources_ <ins>and _rangeMappings_</ins>.
           1. Set _state_.[[GeneratedLine]] to _state_.[[GeneratedLine]] + 1.
           1. Set _state_.[[GeneratedColumn]] to 0.
-          1. Perform DecodeMappingsField of |LineList| with arguments _state_, _mappings_, _names_ and _sources_.
+          1. <ins>Set _state_.[[MappingIndex]] to 0.</ins>
+          1. Perform DecodeMappingsField of |LineList| with arguments _state_, _mappings_, _names_<ins>,</ins> <del>and</del> _sources_ <ins>and _rangeMappings_</ins>.
```

Comment:
I think we should leave `DecodeMappingsField` alone, and instead pass `mappings` to `DecodeRangeMappings`. It allows us define error cases for 
- range mapping point to a `GenColumn` sourceless mapping
- range mapping points to undefined mapping (line too large or index too large)